features/filesystem/fat/FATFileSystem.h

```diff
@@ -35,13 +36,46 @@ using namespace mbed;
  */
 class FATFileSystem : public FileSystemLike {
```

Comment:
This header file could contain informatin about thread safety as our API do (I noticed it is as there are lock/unlock methods)